rudderstack/integrations/destinations/common_config_meta_test.go

```diff
@@ -0,0 +1,199 @@
+// write unit tests for the common_config_meta.go file
+
+package destinations
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	c "github.com/rudderlabs/terraform-provider-rudderstack/rudderstack/configs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetCommonConfigMeta(t *testing.T) {
+	testCases := []struct {
+		description          string
+		supportedSourceTypes []string
+		expectedProperties   []c.ConfigProperty
+		expectedSchema       map[string]*schema.Schema
+	}{
+		{
+			description:          "Valid list of supported source types",
+			supportedSourceTypes: []string{"web", "android", "ios"},
+			expectedProperties: []c.ConfigProperty{
+				c.ArrayWithStrings("oneTrustCookieCategories.web", "oneTrustCookieCategory", "onetrust_cookie_categories.0.web"),
+				c.ArrayWithStrings("oneTrustCookieCategories.android", "oneTrustCookieCategory", "onetrust_cookie_categories.0.android"),
+				c.ArrayWithStrings("oneTrustCookieCategories.ios", "oneTrustCookieCategory", "onetrust_cookie_categories.0.ios"),
+			},
+			expectedSchema: map[string]*schema.Schema{
+				"onetrust_cookie_categories": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "Specify OneTrust category IDs.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"web": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+							"android": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+							"ios": {
+								Type:        schema.TypeList,
+								Optional:    true,
+								Elem:        &schema.Schema{Type: schema.TypeString},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			description:          "Empty list of supported source types",
```

Comment:
I'm not sure we should allow for empty list of supported source types. If not source types are supported, there is nothing the user should configure in respect to onetrust.

Reply:
Addressed.